### DISCUSSION TITLE
Dependency injection in hooks

### DISCUSSION THREAD

**Discussion by MostafaEsmaeili** [DeeJayTC/net-dynamic-api#39] [Show and tell]
It would be beneficial if the library could support dependency injection in hooks, as it would make it much easier to manage dependencies and improve the overall design of the code. or at least all hooks access to `IServiceProvider` to resolve dependencies. 

Is it possible to add support for dependency injection in hooks? If so, I would greatly appreciate it if this could be implemented in a future release of the library.

Thank you for your hard work on this project and I look forward to hearing back from you.

**Comment by DeeJayTC** [comment]
Hi @MostafaEsmaeili thats actually already possible despite that this part could really need a better documentation. 
Have a look at this:

![image](https://user-images.githubusercontent.com/4077759/219904266-550e6a98-4270-4cba-8dcd-8196d1cedf21.png)

the data object you get in all the hooks not only carries the GenericDataContext to do custom queries but also the HTTPContext which exposes a RequestServices function effectively giving you the IServiceCollection you can use to get services. 

Does that  help you?

**Comment by MostafaEsmaeili** [comment]
Hi, @DeeJayTC.  Yes, that is what I am looking for.  Thank you for your response.
